lib/protobuf/message.rb

```diff
@@ -26,11 +26,15 @@ def self.to_json
     # Constructor
     #
 
-    def initialize(fields = {})
+    def initialize(fields = {}, &blk)
       @values = {}
 
-      fields.to_hash.each_pair do |name, value|
-        self[name] = value
+      if blk
+        blk.call(self)
+      else
+        fields.to_hash.each_pair do |name, value|
```

Comment:
I would prefer that we always utilize the given fields hash (if any given), in addition to calling the block. Perhaps:

```ruby
def initialize(fields = {}, &block)
  fields.to_hash.each_pair do |name, value|
    self[name] = value
  end
  block.call(self) if block_given?
end
```

Which would enable you to do this:

```ruby
foo = Foo.new(:bar => 1) do |f|
  f.baz = 2
end
```